### PR TITLE
Use accept-encoding for repodata

### LINKED ISSUE #2807: Consider allowing for conda compression that uses deflate rather than bzip2

### PR DESCRIPTION
**Pull request by mcg1969**
closes #2807, #2794, #2802

### CODE REVIEW

**Comment by kalefranz** [comment]
This is more invasive than I think we want right now.  Let me take a crack at it.  I'll add on top or your commit...

**Comment by mcg1969** [comment]
By all means, take this one over, I was just curious.

File: conda/fetch.py [hunk 1]
```diff
@@ -91,31 +82,48 @@ def fetch_repodata(url, cache_dir=None, use_cache=False, session=None):
...
+    if not ssl_verify:
+        try:
+            from requests.packages.urllib3.connectionpool import InsecureRequestWarning
+        except ImportError:
+            pass
+        else:
+            warnings.simplefilter('ignore', InsecureRequestWarning)
+
+    session = session or CondaSession()
+
```
Comment (kalefranz) [inline conda/fetch.py]: ^^ Just switched the ordering here.  Don't need this logic in front of `if use_cache: return cache`.

File: conda/fetch.py [hunk 2]
```diff
@@ -91,31 +82,48 @@ def fetch_repodata(url, cache_dir=None, use_cache=False, session=None):
...
     headers = {}
     if "_etag" in cache:
         headers["If-None-Match"] = cache["_etag"]
     if "_mod" in cache:
         headers["If-Modified-Since"] = cache["_mod"]
+    if 'anaconda.org' in url:
+        headers['Accept-Encoding'] = 'gzip;q=1.0, identity;q=0.1'
```
Comment (kalefranz) [inline conda/fetch.py]: only gzip, no compress or deflate (although maybe I should add back in deflate?)

**Comment by kalefranz** [comment]
Anything look out of place @mcg1969?

File: conda/fetch.py (hunk 2 above)
Comment (mcg1969) [inline conda/fetch.py]: I personally would do the opposite here. That is: I would do `gzip, deflate, compress, identity` (no quality parameters) for everything *but* `continuum.io`.

Comment (mcg1969) [inline conda/fetch.py]: (But you're the boss :-))

**Comment by mcg1969** [comment]
Looks good to me, with a couple of comments above

**Comment by kalefranz** [comment]
@srossross @stephenakearns This is going to be included in conda 4.1.3 coming out tomorrow...

**Comment by mcg1969** [comment]
For future consideration: remove the hardcoding of `anaconda.org`, replacing it with whatever domain name comes out of `get_binstar`

File: conda/fetch.py [hunk 3]
```diff
@@ -91,24 +81,43 @@ def fetch_repodata(url, cache_dir=None, use_cache=False, session=None):
...
+    headers = {}
     if "_etag" in cache:
         headers["If-None-Match"] = cache["_etag"]
     if "_mod" in cache:
         headers["If-Modified-Since"] = cache["_mod"]
+    if 'anaconda.org' in url:
+        headers['Accept-Encoding'] = 'gzip'
+        filename = 'repodata.json'
+    else:
+        filename = 'repodata.json.bz2'
```
Comment (oleyka) [inline conda/fetch.py]: that was... unexpected. but we survived it :)

**Pull request merged**